routes/web/project/location/vm.rb

```diff
@@ -30,9 +30,10 @@ class CloverWeb
             Validation.validate_port_range(r.params["port_range"])
           end
 
+          parsed_cidr = Validation.validate_cidr(r.params["cidr"])
           pg_range = Sequel.pg_range(port_range.first..port_range.last)
 
-          vm.firewalls.first.insert_firewall_rule(r.params["cidr"], pg_range)
+          vm.firewalls.first.insert_firewall_rule(parsed_cidr.to_s, pg_range)
```

Comment:
(No need to change, just a note for future readers) When the `firewalls.first` is used while inserting a firewall rule, it causes reader to think that change can affect a random private subnet. Though, we don't support multiple subnets for a VM, so it is ok.